### PR TITLE
[ASCII-2428] Create a metaCollector singleton

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

Create a metaCollector singleton

### Motivation

That way we no longer need to provide an optional workloadmeta when calling the tagger functions

This would allow us to fully decouple the different tagger implementation into separate packages, and ensuring the remote tagger implementation do not have a reference to the workloadmeta component

The last three commits also refactor a few places in the codebase that there were passing an optional workloadmeta. component to get the `MetaCollector`

- The workloadmeta component 
- Dogstatsd component
- The `pkg/flare` 

#### The future 
Ideally the `util/containers/metrics` should become **its own component**. That would avoid the usage of globals. Creating the component is a much bigger change. This change unblocks us from splitting the tagger component into multiple implementations, but ideally, we should create a separate component and pass it as a dependency to the places that used it. 

### Describe how to test/QA your changes

Unit tests are updated 

[TBD] How to to test origin detection mode

### CODE REVIEW

File: cmd/dogstatsd/subcommands/start/command_test.go [hunk 1]
```diff
@@ -8,18 +8,19 @@ package start
...
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
 )
 
 func TestStartCommand(t *testing.T) {
 	fxutil.TestOneShotSubcommand(t,
 		[]*cobra.Command{MakeCommand("defaultLogFile")},
-		[]string{"start", "--cfgpath", "PATH"},
+		[]string{"start"},
```
Comment (vickenty) [inline cmd/dogstatsd/subcommands/start/command_test.go:21]: Is this change really necessary for workloadmeta decoupling?

File: comp/dogstatsd/server/parse.go [hunk 2]
```diff
@@ -81,7 +79,7 @@ func newParser(cfg model.Reader, float64List *float64ListPool, workerNum int, wm
 		readTimestamps:   readTimestamps,
 		float64List:      float64List,
 		dsdOriginEnabled: cfg.GetBool("dogstatsd_origin_detection_client"),
-		provider:         provider.GetProvider(wmeta),
+		provider:         provider.GetMetaCollector(),
```
Comment (vickenty) [inline comp/dogstatsd/server/parse.go:82]: This seems unnecessarily fragile. Before this change the provider was made sure to be initialized whenever it was used, now this depends on the code scattered around `cmd` for this. (`GetProvider` starts a background process that feeds data into the meta collector, but `GetMetaCollector` does not).  I get the need to decouple the tagger component from the workloadmeta, but is there really any need to decouple the dogstatsd server?

**Review by ichinaski — APPROVED** [review]
